scripts/infrastructure/roles/manage-non-mgmt-aws-policies.sh

```diff
@@ -0,0 +1,115 @@
+#!/bin/bash
+
+function _substitute_environment_variables() {
+  eval "cat << EOF
+$(cat $1)
+EOF"
+}
+
+AWS_REGION_NAME="eu-west-2"
+
+ACCOUNT_ID=$(aws sts get-caller-identity | jq -r .Account)
+
+#
+# Check we're not running this against MGMT
+#
+. "./scripts/aws/helpers.sh"
+if _validate_current_account "MGMT"; then
+  echo "Please login to non-mgmt profile before running this script"
+  exit 1
+fi
+
+function _update_policy() {
+  policy_name="$1"
+  policy_type="$2"
+  role_names=("${@:3}")
+  aws iam get-policy --policy-arn "arn:aws:iam::${ACCOUNT_ID}:policy/${policy_name}" &> /dev/null
+  if [ $? != 0 ]; then
+    tf_policy=$(_substitute_environment_variables ./scripts/infrastructure/policies/${policy_type}-policy.json)
+    aws iam create-policy \
+      --policy-name "${policy_name}" \
+      --policy-document "${tf_policy}" \
+      --region "${AWS_REGION_NAME}" \
+      || exit 1
+
+    for role in "${role_names[@]}"; do
+      if [ "${role}" == "NHSDevelopmentRole" ]; then
+        if ! _validate_current_account "PROD"; then
+          aws iam attach-role-policy \
+            --role-name "${role}" \
+            --policy-arn "arn:aws:iam::${ACCOUNT_ID}:policy/${policy_name}" \
+            --region "${AWS_REGION_NAME}" \
+            || exit 1
+        fi
+      else
+        aws iam attach-role-policy \
+          --role-name "${role}" \
+          --policy-arn "arn:aws:iam::${ACCOUNT_ID}:policy/${policy_name}" \
+          --region "${AWS_REGION_NAME}" \
+          || exit 1
+      fi
+    done
+  else
```

Comment:
Does this actually update the policy? From what I can tell this section runs if the policy does exist but the code seems to attach that role again but I cant tell if it has redeployed the new one?

Reply:
Or does updating the version of the policy later update it for all Roles that are using it?